ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -802,7 +802,21 @@ let on_notification server (notification : Client_notification.t) :
        after receiving this notification *)
     let configuration = Configuration.update state.configuration req in
     Fiber.return { state with configuration }
-  | DidSaveTextDocument _
+  | DidSaveTextDocument { textDocument = { uri }; _ } -> (
+    let open Fiber.O in
+    let state = Server.state server in
+    match Document_store.get_opt state.store uri with
+    | None ->
+      ( Log.log ~section:"on receive DidSaveTextDocument" @@ fun () ->
+        Log.msg "saved document is not in the store" [] );
+      Fiber.return state
+    | Some doc ->
+      (* we need [update_text] with no changes to get a new merlin pipeline;
+         otherwise the diagnostics don't get updated *)
+      let* doc = Document.update_text doc [] in
+      Document_store.put store doc;
```

Comment:
What I meant was to move the code in `DidChangeDocument` into a function and just call it in both clauses. This is good as well however.